tests/test_tls.py

```diff
@@ -1420,6 +1402,32 @@ def test_verify_subject_with_subjaltname(self):
                 )
             self.assertEqual(
                 str(cm.exception),
-                "hostname 'acme.com' doesn't match either of '*.example.com', "
-                "'example.com'",
+                "hostname 'acme.com' doesn't match either of "
+                "DNSPattern(pattern=b'*.example.com'), DNSPattern(pattern=b'example.com')"
+            )
+
+    def test_verify_subject_with_subjaltname_ipaddress(self):
+        certificate, _ = generate_ec_certificate(
+            alternative_names=["1.2.3.4"],
+            common_name="1.2.3.4",
+        )
+        cadata = certificate.public_bytes(serialization.Encoding.PEM)
+
+        with patch("aioquic.tls.utcnow") as mock_utcnow:
+            mock_utcnow.return_value = certificate.not_valid_before
+
+            # valid
+            verify_certificate(
+                cadata=cadata, certificate=certificate, server_name="1.2.3.4"
```

Comment:
As explained, `server_name` is not meant to hold an IP address:
- search through tls.py and you'll see it's being used for SNI.
- look at asyncio/client.py and you'll see the `QuicConfiguration.server_name` is set to `None` if the host is an IP address